### DISCUSSION TITLE
Problems with Tycho workspace via Oomph

### DISCUSSION THREAD

**Discussion by ghost** [eclipse-tycho/tycho#1187] [Q&A]
I've just checked out a fresh Tycho workspace as described in the contributing guidelines, but I'm unable to resolve all of the project errors.

The first issue I noticed is that the installer seems to get stuck during the *Process resource updates* phase at around 78%, where it seems to scan the same folders over and over again (I at least saw tycho-core pop up multiple times). After a few minutes, I interrupted this task as it didn't go anywhere.

![Capture](https://user-images.githubusercontent.com/92990747/180714416-8bfc6224-49ed-4d14-9c8e-5b33b6d13311.PNG)

The remaining tasks are able to complete without problem, but I end up with a broken workspace. 

![Capture2](https://user-images.githubusercontent.com/92990747/180715206-59675a33-4f60-4571-bedd-794288fe2de8.PNG)

What did I do wrong?

I can provide the error messages, if required. But given that I haven't done anything out of the ordinary, I think the error always shows up, when setting up the workspace.

**Comment by laeubi** [comment]
> The first issue I noticed is that the installer seems to get stuck during the _Process resource updates_ phase at around 78%

This is sadly an m2e issue, so you can either install m2e snapshot or cancel the stuck job.

> The remaining tasks are able to complete without problem, but I end up with a broken workspace.

Sadly the setup is a bit flawed today (as no one really updates it), even if it sound strange, you should be able to ignore the errors and just start coding, as Tycho can't be run out of the IDE itself it won't harm much.

> What did I do wrong?

Nothing specific, I hope that with new m2e version and some of the planned cleanups we might get a more smooth user-experience using the setup soon. Maybe @HannesWell also would jump in as he already has fixed the m2e setup a while ago...

**Reply by HannesWell** [reply]
I also encountered many similar errors at the Weekend, but didn't have the time to fix it. The cause seemed to be that for the tycho-bundle-plugins the .classpath file was not generated by M2E and the classes in those Projects are there not recogniced and compiled. @pzi-dsa can you verify that?

Explicitly calling Maven -> 'Update Project Config', did not help as well and even deleting such project from the workspace and importing ist as an existing Maven project did not solve the issue for me too.
But maybe IT works for you.

Unfortunately I'm currently on the train for a two week vacation and can not debug/fix this in M2E before returning.
If you @pzi-dsa or @laeubi want to address this issue in M2E before (given that M2E ist really causing it), I suggest you start the 'Eclipse-M2E-IDE' Launch Config in the M2E workspace (ignore the warning about xml.resolver) and import one of the affected Tycho projects ans see what happens and how the import can be fixed.

**Reply by laeubi** [reply]
Might be related to https://github.com/eclipse-m2e/m2e-core/issues/577

**Reply by ghost** [reply]
> I also encountered many similar errors at the Weekend, but didn't have the time to fix it. The cause seemed to be that for the tycho-bundle-plugins the .classpath file was not generated by M2E and the classes in those Projects are there not recogniced and compiled. 

That seems to be exactly it. Both the .classpath file as well as the Java natures are missing in those projects.

> Explicitly calling Maven -> 'Update Project Config', did not help as well and even deleting such project from the workspace and importing ist as an existing Maven project did not solve the issue for me too. But maybe IT works for you.

Nope. No success with that, either. Unless I want to manually patch the problematic projects.

If I have time, I can perhaps look into it. Though I'm already busy with my current task for which I need Tycho to do some remote debugging. That should still be possible, even with compile errors.

Thank both of you for your quick response!

**Reply by laeubi** [reply]
> Though I'm already busy with my current task for which I need Tycho to do some remote debugging. That should still be possible, even with compile errors.

Yep, in that case it dosen't matter .. just make sure that the projects you want to debug are imported as java projects to have source lookups.
